### PR TITLE
Sanitize report's inputs and usernames - 3.13.x

### PR DESCRIPTION
**Pull request by Desvelao**
# Description

This PR adds parameter validation to the report's endpoints.

Closes https://github.com/wazuh/wazuh-kibana-app/issues/4329

# Changes

- Add parameter validation to the reporting endpoints.
- Removed some parameters in the reporting endpoints.
- Build the report name in the backend.
- Add testing with Jest.
- Add tests with invalid parameters in the reporting endpoints.

# Test

- Do requests to the reports endpoints with unexpected / forbidden parameters using some tools such as cURL or Postman.
- Check that each kind of report can be generated as expected:
  - Modules (with an agent selected, without an agent selected, changing the type of selected time range, etc...)
  - Group
  - Agent configuration
  - Agent inventory

### CODE REVIEW

**Comment by Desvelao** [comment]
> Add the MD5 dependency with:
> 
> ```shell
> npm install md5 --save
> ```

# Changes
Add the project dependency of `md5`.

**Comment by Desvelao** [comment]
> Add the MD5 dependency with:
> 
> ```shell
> npm install md5 --save
> ```

The changes included in this PR don't require the `md5` library.

File: server/controllers/wazuh-reporting.js [hunk 1]
```diff
@@ -1895,6 +1896,20 @@ export class WazuhReportingCtrl {
         const isAgentConfig = tab === 'agentConfig';
         const isGroupConfig = tab === 'groupConfig';
 
+        // Generate the filename of report depeding on request parameters
+        const filename = tab === 'syscollector'
+          ? `wazuh-agent-inventory-${agentID}-${this.generateReportTimestamp()}.pdf`
```
Comment (asteriscos) [inline server/controllers/wazuh-reporting.js:1901]: Here you are expecting the ID of the agent in `agentID`, but actually the react service sends the agent ID in the `isAgents` parameter. 

This is the request when clicking Agents / Inventory PDF report
image

image

**Review by asteriscos — CHANGES_REQUESTED** [review]
Some request parameters are not sent as they should. Please review the comments.

File: server/controllers/wazuh-reporting.js (hunk 1 above)
Comment (Desvelao) [reply server/controllers/wazuh-reporting.js]: Good catch! It is fixed now.

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark: 

Screenshot from 2022-08-11 16-51-28

image

image

**Review by matiasmoreno876 — APPROVED** [review]
LGTM

**Pull request merged**